lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,206 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache qw(STATUS_PROCESSED STATUS_ENQUEUED STATUS_DOWNLOADING STATUS_IGNORE STATUS_ERROR);
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+use constant DEFAULT_MINION_WORKERS => 5;
+
+BEGIN { srand(time) }
+
+my $_token;
+
+sub _gen_session_token { $_token = int(rand(999999999999)) }
+
+my $enqueued = Mojo::Collection->new;
+app->hook(
+    before_server_start => sub {
+        _gen_session_token();
+        $enqueued = Mojo::Collection->new;
+    });
+
+plugin Minion => {SQLite => 'sqlite:' . OpenQA::Worker::Cache->from_worker->db_file};
+plugin 'Minion::Admin';
+plugin 'OpenQA::Worker::Cache::Task::Asset';
+plugin 'OpenQA::Worker::Cache::Task::Sync';
+
+sub SESSION_TOKEN { $_token }
```

Comment:
It's weird to name `SESSION_TOKEN` like a constant when it isn't. Since it's a class variable maybe name it `$Token` above and just use that directly. (Yes, that capitalization is the common convention for class scoped variables 😉)

Reply:
Iit is a function because during a first draft it was meant to be a constant, but to ease testing was changed to a package scoped variable.